VAMobile/src/components/PhotoUpload.tsx

```diff
@@ -0,0 +1,171 @@
+}
+
+type StyledImageProps = {
+  /** prop to set image width */
+  width: number
+  /** prop to set image height */
+  height: number
+  /** Hardcoded radius of 50 due to circular design plan */
+  borderRadius: number
+  /** Hardcoded radius of 2 due to design plan */
+  borderWidth: number
+  /** Hardcoded border color of white */
+  borderColor: string
+}
+
+const StyledImage = styled(Image)<StyledImageProps>`
+  width: ${themeFn<StyledImageProps>((_theme, props) => props.width)}px;
+  height: ${themeFn<StyledImageProps>((_theme, props) => props.height)}px;
+  border-radius: ${themeFn<StyledImageProps>((_theme, props) => props.borderRadius)}px;
+  border-width: ${themeFn<StyledImageProps>((_theme, props) => props.borderWidth)}px;
+  border-color: ${themeFn<StyledImageProps>((_theme, props) => props.borderColor)};
+`
+//TODO: Add this back to the VeteranStatusScreen when decided upon
+const PhotoUpload: FC<PhotoUploadProps> = ({ width, height }) => {
+  const { t } = useTranslation(NAMESPACE.COMMON)
+  const confirmAlert = useDestructiveActionSheet()
+  const photoUploadBorderRadius = 50
+  const photoUploadBorderWidth = 2
+  const showActionSheetWithOptions = useShowActionSheet()
+  const VETERAN_STATUS_PHOTO = '@store_veteran_status_photo'
+  const [uri, setUri] = useState('')
+  const options = [t('fileUpload.camera'), t('fileUpload.photoGallery'), t('cancel')]
+  const uploadBorderColor = theme.colors.border.photoUpload
+
+  const getPhotoFromStorage = async (): Promise<void> => {
+    const override = await AsyncStorage.getItem(VETERAN_STATUS_PHOTO)
+    if (override !== null) {
+      setUri(override)
+    }
+  }
+
+  useEffect(() => {
+    if (uri === '') {
+      getPhotoFromStorage()
+    }
+  }, [uri])
+
+  const photo = (): ReactNode => {
+    return (
+      <StyledImage source={{ uri }} width={width} height={height} borderRadius={photoUploadBorderRadius} borderWidth={photoUploadBorderWidth} borderColor={uploadBorderColor} />
+    )
+  }
+
+  const uploadCallback = (response: ImagePickerResponse): void => {
+    const { assets, errorMessage, didCancel } = response
+    if (didCancel) {
+      return
+    } else if (errorMessage) {
+      //TODO Snackbar in this case? Discuss error cases
+      return
+    } else {
+      if (assets?.length === 0) {
+        return
+      }
+      if (assets && assets[0] && assets[0].uri) {
+        setUri(assets[0].uri)
+        AsyncStorage.setItem(VETERAN_STATUS_PHOTO, assets[0].uri)
+      }
+    }
+  }
+
+  const onPress = (): void => {
+    if (uri) {
+      confirmAlert({
+        title: t('removePhoto'),
+        cancelButtonIndex: 0,
+        destructiveButtonIndex: 1,
+        buttons: [
+          {
+            text: t('keep'),
+            onPress: () => {},
+          },
+          {
+            text: t('remove'),
+            onPress: () => {
+              setUri('')
+              AsyncStorage.setItem(VETERAN_STATUS_PHOTO, '')
```

Comment:
Do we need to show confirmation that the photo has been removed?

Reply:
Nope.